docs/DeveloperGuide.md

```diff
@@ -1,15 +1,247 @@
+
+**5. DataReader**
+
+DataReader class reads two external .txt files to acquire the list of Partner Universities and list
+of Modules available in the PUs, and provides this information to other components.
+
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Storage
+
+The Storage class handles the loading and storing of information into a single text file in the User's computer.
+During the initialisation of the Storage class, which is at the start of the program, loading of data occurs.
+From here on, querying of the Storage class is allowed, where modules saved by the user can be accessed through
+the ListCurrentCommand. There are only two commands that will cause changes to the text file. They are the AddCommand
+and DeleteCommand. This involves adding of new modules and deleting of old modules to the Storage. The text file will
+be continuously updated every time an Add or Delete command is called.
+
+How module data is stored in text file:
+
+Module information is stored in one single line separated by commas
+`univID`,`moduleCode`,`moduleName`,`moduleMCs`,`nusModuleCode`,`nusModuleName`,`nusModuleMcs`
+
+Sequence Diagram of Storage initialisation:
+
+![Storage.png](diagrams%2FStorage%2FStorage.png)
+
+The Storage class also handles the adding of new modules and the deleting of past modules. When any of this occurs, the
+txt file will be updated immediately after the successful adding/deletion of saved modules in the Storage.
+
+- For adding of newly saved modules, they are added through appending to the saved_modules.txt file
+- For deleting of past saved modules, they are deleted, and the txt file is updated by rewriting every module from the
+  ArrayList of saved modules
+
+### Parser
+
+The parser class is responsible for parsing the user's input commands and returning the appropriate command object.
+The commands that the parser class will initialise are ListPuCommand(), ListCurrentCommand(modules),
+prepareListPuModulesCommand(userCommandSecondKeyword, universities), ExitCommand(),
+prepareAddModuleCommand(storage, userCommandSecondKeyword, puModules, universities),
+DeleteModuleCommand(storage, indexToRemove, modules), and HelpCommand(). The parser class will handle error checking by
+throwing InvalidCommandException if the user's input command does not match the specified format.
+
+The following class diagrams illustrates the relationship between the Parser class and the Command classes.
+- (TODO: finish up the rest of the command cases)
+![ParserSequenceDiagram.png](diagrams%2FParserSequenceDiagram.png)
+
+### Help Command
+
+The help command provides a list of commands and the commands' respective input format for the user.  
+> Syntax: /help
+
+The following sequence diagram shows the relationship between the classes involved when the delete command is called.
+
+![HelpCommandSequenceDiagram.png](diagrams%2FHelpCommandSequenceDiagram.png)
```

Comment:
The method call printHelpCommandMessage() from :HelpModuleCommand should be at the start of the activation bar.